plugins/mcreator-core/blockly/js/field_data_list_selector.js

```diff
@@ -2,27 +2,35 @@
  * This class represents a data list field that can be double-clicked to open a list entry selector.
  * The behaviour is similar to block/item selectors or condition selectors for entity AI blocks
  */
-class FieldDataListSelector extends Blockly.FieldLabelSerializable {
+class FieldDataListSelector extends Blockly.Field {
+
+    EDITABLE = true;
+    SERIALIZABLE = true;
+    CURSOR = 'default';
+
     constructor(datalist = '', opt_validator, opt_config) {
-        super(javabridge.t('blockly.extension.data_list_selector.no_entry'), 'entry-label');
+        super('', opt_validator, opt_config);
         this.type = datalist;
         this.typeFilter = null;
         this.customEntryProviders = null;
-        this.entry = FieldDataListSelector.getDefaultEntry();
-        this.setTooltip(this.getText_());
 
-        this.EDITABLE = true;
+        this.maxDisplayLength = 75;
 
         if (opt_config)
             this.configure_(opt_config);
-        if (opt_validator)
-            this.setValidator(opt_validator);
-    }
 
-    // The default entry is ",No entry selected". Since the value is an empty string, the procedure editor will show a compile error
-    static getDefaultEntry() {
-        return ',' + javabridge.t('blockly.extension.data_list_selector.no_entry');
-    }
+        // Show the full name of the selected value, or the "Double click to select value" message
+        let thisField = this;
+        this.setTooltip(function () {
+            return thisField.getValue() ?
+                thisField.readableName : javabridge.t('blockly.field_data_list_selector.tooltip.empty');

```

Comment:
I'd split this in two lines to follow code style in other files (did not check other JS files, maybe it's not followed here entirely):
```suggestion
                thisField.readableName :
                javabridge.t('blockly.field_data_list_selector.tooltip.empty');
```

Reply:
Fair